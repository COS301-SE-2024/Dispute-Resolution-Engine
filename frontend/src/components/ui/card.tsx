import * as React from "react";

import { cn } from "@/lib/utils";
import { forwardSlot } from "../util";

const Card = forwardSlot<HTMLDivElement>(
  "Card",
  "div",
  "bg-surface-light-50 dark:bg-surface-dark-800 rounded-lg dark:border dark:border-primary-500/10 max-w-5xl mx-auto shadow-md"
);

const CardHeader = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(
  ({ className, ...props }, ref) => (
    <div ref={ref} className={cn("flex flex-col space-y-1.5 px-6 pt-6", className)} {...props} />
  )
);
CardHeader.displayName = "CardHeader";

const CardTitle = React.forwardRef<HTMLParagraphElement, React.HTMLAttributes<HTMLHeadingElement>>(
  ({ className, ...props }, ref) => (
    <h3
      ref={ref}
      className={cn("text-2xl font-semibold leading-none tracking-tight", className)}
      {...props}
    />
  )
);
CardTitle.displayName = "CardTitle";

const CardDescription = React.forwardRef<
  HTMLParagraphElement,
  React.HTMLAttributes<HTMLParagraphElement>
>(({ className, ...props }, ref) => (
  <p ref={ref} className={cn("text-sm text-slate-500 dark:text-slate-400", className)} {...props} />
));
CardDescription.displayName = "CardDescription";

const CardContent = forwardSlot<HTMLDivElement>("CardContent", "div", "p-6");
const CardFooter = forwardSlot<HTMLDivElement>("CardFooter", "div", "flex items-center p-6 pt-0");

export { Card, CardHeader, CardFooter, CardTitle, CardDescription, CardContent };
