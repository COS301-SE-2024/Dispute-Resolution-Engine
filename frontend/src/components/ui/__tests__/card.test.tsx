import renderer from 'react-test-renderer';
import { Card } from '../card';

it('renders correctly', () => {
  const tree = renderer
    .create(<Card />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});