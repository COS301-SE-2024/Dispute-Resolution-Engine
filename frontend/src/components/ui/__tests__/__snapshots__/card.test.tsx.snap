// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-surface-light-50 dark:bg-surface-dark-800 rounded-lg dark:border dark:border-primary-500/10 max-w-5xl mx-auto shadow-md"
/>
`;
